### PR TITLE
Status info bar

### PR DESCRIPTION
**Pull request by frankfu03**

### CODE REVIEW

File: src/client/app/shared/status-info/status-info.component.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+                auth = auth.toUpperCase();
+                if (auth === 'SUPERUSER') {
+                    return true;
+                } else if (auth.startsWith('EDIT-') && auth.slice(5) === this.siteId) {
+                    return true;
+                }
+            }
+            return false;
+        }
+    }
```
Comment (lbodor) [inline src/client/app/shared/status-info/status-info.component.ts]: If you extend `AbstractBaseComponent`, you will have access to `isEditable()`, which I think does the same thing. Alternatively, you can inject `UserAuthService` and call `hasAuthorityToEditSite()` yourself.

**Comment by lbodor** [comment]
:+1:

File: src/client/app/shared/status-info/status-info.component.html [hunk 2]
```diff
@@ -0,0 +1,20 @@
+<div class="status-info-bar">
```
Comment (oehm-smith) [inline src/client/app/shared/status-info/status-info.component.html]: Good one making this a separate component

File: src/client/app/shared/status-info/status-info.component.html [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+            any sites
+        </span>
+        <span *ngIf="siteId != null">
+            {{siteId}}. {{getFormStatus()}}
+        </span>
+    </span>
+    <span *ngIf="user != null">
+        <span>You have logged in as "{{user.profile.sub}}"</span>
+        <span *ngIf="siteId == null">
+            and are authorised to edit {{getUserAuthorityString()}})
```
Comment (oehm-smith) [inline src/client/app/shared/status-info/status-info.component.html]: You need to add a period '.'

File: src/client/app/shared/status-info/status-info.component.ts (hunk 1 above)
Comment (frankfu03) [inline src/client/app/shared/status-info/status-info.component.ts]: Good suggestion. As UserAuthService has been already injected into StatusInfo Component, I choose to call userAuthService.hasAuthorityToEditSite() inside isAuthorisedSite().

File: src/client/app/shared/status-info/status-info.component.ts [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+        for (let auth of this.user.profile.authorities) {
+            auth = auth.toLowerCase();
+            if (auth === 'superuser') {
+                return 'all sites';
+            } else if (auth.startsWith('edit-')) {
+                authorities.push(auth.slice(5).toUpperCase());
+            }
+        }
+        return authorities.join();
+    }
```
Comment (frankfu03) [inline src/client/app/shared/status-info/status-info.component.ts]: Removed.

File: src/client/app/shared/status-info/status-info.component.html (hunk 3 above)
Comment (frankfu03) [inline src/client/app/shared/status-info/status-info.component.html]: Yes, it is done. Thanks.

File: src/client/app/shared/status-info/status-info.component.ts (hunk 4 above)
Comment (frankfu03) [inline src/client/app/shared/status-info/status-info.component.ts]: I moved this method to UserAuthService, so both StatusInfo and Toolbar components can call this method.

**Pull request merged**